### PR TITLE
docs: Bluetooth: Add sections on GTBS and GMCS

### PR DESCRIPTION
**Pull request by Thalley**
Since GTBS and GMCS in both spec definitions and
in the Zephyr implemenation are notably different, it is worthwhile pointing these out in the documentation for content control.

### CODE REVIEW

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 1]
```diff
@@ -313,6 +313,110 @@ The MCP implementation supports the following roles
...
+   There is no specified manner of representing a characteristic from each individual TBS that
+   resides on the device to the same characteristic of the GTBS.
+
+   For example, if there is more than one TBS on a device and each has a unique telephone bearer
+   name (e.g., Name1 and Name2),
+   the way the GTBS represents the telephone bearer name is left up to the implementation.
+   GTBS is suited for clients that do not need to access or control all the
+   information available on specific telephone bearers.
+
+This means that a GTBS instance represent one or more telephone bearers.
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:345]:
```suggestion
This means that a GTBS instance represents one or more telephone bearers.
```

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 2]
```diff
@@ -313,6 +313,110 @@ The MCP implementation supports the following roles
...
+A telephone bearer could be any application on a device that can handle (telephone) calls,
+such the default Call application on a smartphone,
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:347]:
```suggestion
such as the default Call application on a smartphone,
```

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 3]
```diff
@@ -313,6 +313,110 @@ The MCP implementation supports the following roles
...
+In Zephyr the implementation of GBTS is that it contains some generic information,
+such as the provider name which is defined to  simply be "Generic TBS",
+but the majority of the information in the GTBS instance in Zephyr has been implemented to be a
+union of the data of the other bearers.
+For example if you have a bearer for regular phone calls and
+Teams and have an active call in both bearers,
+then each of those bearers will report a single call,
+but the GTBS instance will report 2 calls,
+making it possible for a simple Call Control Client to control all calls from a single bearer.
+Similarly the supported URIs for each bearer at also made into a union in GTBS, and when placing
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:361]:
```suggestion
Similarly the supported URIs for each bearer are also made into a union in GTBS, and when placing
```

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 4]
```diff
@@ -313,6 +313,110 @@ The MCP implementation supports the following roles
...
+a call using the GTBS the server will pick the most suited bearer depending on the UCI.
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:362]:
```suggestion
a call using the GTBS the server will pick the most suited bearer depending on the URI.
```

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 5]
```diff
@@ -313,6 +313,110 @@ The MCP implementation supports the following roles
...
+For example calls with URI `tel` would go to the regular phone application,
+and calls with the URI `skype` would go to the Teams application.
+
+In conclusion the GTBS implementation in Zephyr is a union of the non-generic telephone bearers.
+
+Generic MCS
+^^^^^^^^^^^
+
+The MCS spec defineds GMCS as
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:371]:
```suggestion
The MCS spec defines GMCS as
```

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 6]
```diff
@@ -313,6 +313,110 @@ The MCP implementation supports the following roles
...
+   the same meaning applies to GMCS as well.
+
+This means that a GMCS instance works the same way as an MCS instance,
+and it follows that GMCS
+
+   controls the media playback for a specific media player within the device
+
+A media player on a device could be anything that plays media,
+such as a Spotify or Youtube application on a smartphone.
+Thus if a device have multiple MCS instances,
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:394]:
```suggestion
Thus if a device has multiple MCS instances,
```

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 7]
```diff
@@ -313,6 +313,110 @@ The MCP implementation supports the following roles
...
+The MCS spec does however provide an example of GMCS where a device can
+
+   Implement a GMCS that provides status and control of media playback for the device as a whole.
+
+Which may indicate that an implementation may use GMCS to represent all media players with GMCS and
+not a specific media player as stated above. In the case where a device does not have any MCS
+instances and only GMCS, then GMCS will point to a generic instance.
+
+The Zephyr implementation of MCS and GMCS is incomplete,
+and currently only supports either instanciating a single instance that can either be a MCS or GMCS.
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:409]:
```suggestion
and currently only supports instantiating a single instance that can be either an MCS or GMCS.
```

File: doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst [hunk 8]
```diff
@@ -313,6 +313,110 @@ The MCP implementation supports the following roles
...
+This means that the implementation is neither complete not spec-compliant.
```
Comment (kruithofa) [inline doc/connectivity/bluetooth/bluetooth-le-audio-arch.rst:410]:
```suggestion
This means that the implementation is neither complete nor spec-compliant.
```

**Review by kruithofa — CHANGES_REQUESTED** [review]
Some minor typos

**Review by kruithofa — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**